test/penalizer/Penalizer.test.ts

```diff
@@ -290,6 +273,81 @@ contract('Penalizer', function ([relayOwner, relayWorker, relayManager, otherAcc
     })
   })
 
+  describe('should receive/reject claims according with the commitment time', function () {
+    before(async function () {
+      await penalizer.setHub(relayHub.address, { from: await penalizer.owner() })
```

Comment:
could you please explain why this line is necessary?

if i'm not mistaken after the execution of the `it` statement on line `70` we shouldn't need to set a hub anymore, but maybe i am missing something

Reply:
I think that the line is required in order to make tests independent from each other. I mean, I believe that the current test shouldn't depend on other tests, but all the required operations should be performed in the `before`/`beforeEach` statements. Ideally, I should be able to run each test alone, without the need of running other tests to satisfy pre-conditions. Maybe we should adjust the other tests to avoid that also. WDYT?

Reply:
> I think that the line is required in order to make tests independent from each other. I mean, I believe that the current test shouldn't depend on other tests, but all the required operations should be performed in the `before`/`beforeEach` statements. Ideally, I should be able to run each test alone, without the need of running other tests to satisfy pre-conditions. Maybe we should adjust the other tests to avoid that also. WDYT?

in principle, i agree entirely. in practice though, it's not what's happening. as it is, line `278` is currently overwriting the hub value with its current value (resulting in no change). this is because `penalizer` is a global variable—available to all tests—as are the other contracts (as well as a couple other variables defined from line `28` onwards). 

while tests should be ideally independent from each other, i'm not sure it's worth the effort of, for example, creating a new environment (line `43`), deploying all the contracts, funding, etc., for each test case that we want to run. although doing this would indeed increase independence, the cost would be execution time, and _possibly_ readability and maintainability.

in the end, we're trying to strike a balance between independence and execution time/code complexity, and i think this can be achieved in more than one way. however, if we want to set the hub from scratch for each test, then we should be also deploying the contracts each time, and this would mean creating a new environment each time too. i think this might be overkill.

Reply:
Yep, I agree that it could be overkill however, I tend to split the initialization code and the business logic of what I want to test. I think it could be acceptable to avoid an expensive initialization for the sake of optimization (time/computational cost), but I would like to be sure that changing a single test doesn't impact the execution of the other tests.
In my opinion, I would prefer to call that method on the initialization and "unset" it just before that single unit test (if possible). I know it's a subjective point of view, so no problem for me if you prefer to remove it from here, I get your point.

Reply:
we had a call with @antomor and agreed to fix this as shown in [this commit](https://github.com/rsksmart/rif-relay/pull/165/commits/18124f3c8d3cab3b79adab31b648e7ca0500f200)